Fix a couple issues

saveas446 opened the pull request:
Fixed enemy collision not working and a line in P_USER.C that was causing a compiler error.

saveas446 committed:
Fix a line in the player movement code

saveas446 committed:
Fix colliding with enemies

SSNTails commented on P_USER.C:
```diff
@@ -297,7 +297,11 @@ void P_MovePlayer (player_t* player)
+
+// Reverted to the SRB2 Xmas 0.96 version of this line based on analysis of the Demo 4.1 EXE 
+// Save 21-05-2025
+	player->speed = (sqrt((abs(player->mo->momx/100000)*abs(player->mo->momx/100000)) + (abs(player->mo->momy/100000)*abs(player->mo->momy/100000))));
```
While this may be v4.1 accurate, Last130 is better represented by Demo v4.35, as it is a copy of v4.1's code plus a few months work after. The Xmas speed measurement is hilariously wrong, and the statements after this have also been adjusted to account for the new, correct speed calculation.

saveas446 replied:
Thanks for letting me know. It's been changed to the version found in the code of the Demo 4.35 EXE.

SSNTails replied:
Updated my comment to reflect that I meant to say this repository is v4.1 plus a few months work after. It's still best represented by v4.35 due to its proximity.. v4.32 was the port to Legacy v1.32.

SSNTails commented on P_MAP.C:
```diff
@@ -781,6 +783,11 @@ if(tmthing->player) // Is the moving/interacting object the player?
+				case MT_REDCRAWLA:
+				case MT_GFZFISH:
+				case MT_JETTBOMBER:
+				case MT_JETTGUNNER:
```
Does this also need MT_SKIM ?
Also pretty floored that hitting enemies isn't working!

saveas446 replied:
For me, enemy collision has been a persistent issue whenever I've worked with this version of the code.
I thought it was just a problem with how I was setting things up, but I found an old build from Tobi and that also had the same bug.

Also, the Skim has been added now. Not sure how I forgot that.

SSNTails reviewed (changes requested):
Thanks for the PR! This is great! Please take a look at my notes.

saveas446 committed:
Fixed Skim

saveas446 committed:
Changed player speed calculation to a restored version of Demo 4.35's

saveas446 committed:
Turns out the function was just P_AproxDistance...

I should've probably checked the references to it to make sure it wasn't just another function being used.

saveas446 committed:
Add the Visual Studio gitignore file

saveas446 committed:
We need the WIN32 folder.
